runtime/compiler/runtime/IProfiler.cpp

```diff
@@ -2955,11 +2955,12 @@ TR_IPBCDataCallGraph::canBeSerialized(TR::PersistentInfo *info)
  * @return void
  */
 void
-TR_IPBCDataCallGraph::serialize(uintptrj_t methodStartAddress, TR_IPBCDataStorageHeader *storage, TR::PersistentInfo *info)
+TR_IPBCDataCallGraph::serialize(TR_J9SharedCache *sharedCache, TR_IPBCDataStorageHeader *storage, TR::PersistentInfo *info)
    {
    TR_IPBCDataCallGraphStorage * store = (TR_IPBCDataCallGraphStorage *) storage;
-   TR_ASSERT(_pc >= methodStartAddress, "_pc=%p should be larger than methodStartAddress=%p\n", (void*)_pc, (void*)methodStartAddress);
-   storage->pc = _pc - methodStartAddress;
+   uintptr_t offset = (uintptr_t)sharedCache->offsetInSharedCacheFromPointer((void *)_pc);
```

Comment:
This won't work unfortunately. I understand why you're making this change, but we don't want to convert the bytecode PC to an SCC offset for JITServer (because the bytecode is not guaranteed to be in the SCC), we want to convert it to an offset from the start of the bytecode itself, which is what the original code was doing.

Unfortunately I changed the `createPersistentCopy()` and `loadFromPersistentCopy()` to take a TR_J9SharedCache* instead of a pointer to the start of some memory to calculate the offset from because a single pointer isn't enough in a multi-SCC environment, so here we are.

The easiest solution is to restore copies of the old versions and just have overloaded versions, one taking a TR_J9SharedCache and the other a single pointer, but I'll have to take a closer look at `TR_IPBCDataCallGraph` which is the most complicated type of profiling info.

Reply:
So i guess for now we should just restore copies of the old versions for JITServer?

Reply:
That's OK for `TR_IPBCDataEightWords::createPersistentCopy()` and `TR_IPBCDataFourBytes::createPersistentCopy()` for but `TR_IPBCDataCallGraph::createPersistentCopy` does a lot more work that shouldn't be duplicated naively.

Can you try factoring this code out into a private helper function? https://github.com/eclipse/openj9/blob/57ec9be1c014201fb79b51c5b39549db69998a0e/runtime/compiler/runtime/IProfiler.cpp#L3050-L3100

That way you can create two overloads of `TR_IPBCDataCallGraph::createPersistentCopy()` that can calculate the PC in different ways but call the helper to do the rest of the work.